src/app/ffbe/character-equipments-display/character-equipments-display.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Component, Input, OnInit, OnChanges} from '@angular/core';
+import {Personnage} from '../model/personnage.model';
+import {Equipment} from '../model/equipment.model';
+import {UniteEquipements} from '../model/unite-equipements.model';
+import {FfchClientService} from '../services/ffch-client.service';
+import {FfbeUtils} from '../utils/ffbe-utils';
+import {isNullOrUndefined} from 'util';
+
+@Component({
+    selector: 'app-character-equipments-display',
+    templateUrl: 'character-equipments-display.component.html',
+    styleUrls: ['character-equipments-display.component.css']
+})
+export class CharacterEquipmentsDisplayComponent implements OnInit {
+
+  @Input() personnage: Personnage;
+  public equipmentErrors: Array<string> = [];
+  public equipmentsFromFfch: Array<Equipment> = [];
+
+  constructor(private ffchClientService: FfchClientService) {
+  }
+
+  ngOnInit() {
+  }
+
+  ngOnChanges() {
+    this.getEquipmentsFromFfch();
+  }
+
+  public getEquipmentsFromFfch() {
+    this.equipmentErrors = [];
+    this.ffchClientService.getUniteEquipementsByUniteNumero$(this.personnage.unites[0].numero)
+    .subscribe(uE => this.equipmentsFromFfch = (isNullOrUndefined(uE) ? null : FfbeUtils.findEquipmentsByFfchIds(uE.equipements_ffch_ids)),
+      error => this.equipmentErrors.push('Erreur lors du traitement de l\'unité ' + this.personnage.unites[0].numero + ' : ' + error));
+
+  }
+
+  public isEquipementPresentInFfchDB() : boolean {
+    return !isNullOrUndefined(this.equipmentsFromFfch) && this.equipmentsFromFfch.length > 0;
+  }
+
+  public isEquipementCorrectInFfchDB() : boolean {
+    return (this.equipmentsFromFfch.length == this.personnage.equipements.length) && this.personnage.equipements.every(equipement => this.equipmentsFromFfch.some(equipmentFromFfch => equipmentFromFfch.ffchId == equipement.ffchId));
+  }
+
+  public isEquipmentErrorsDisplayed(): boolean {
+    return Array.isArray(this.equipmentErrors) && this.equipmentErrors.length > 0;
+  }
+
+  public sendEquipmentsToFfch() {
+    const uniteEquipements = new UniteEquipements(this.personnage.unites[0].numero, this.personnage.equipements.map(equipement => equipement.ffchId));
+    this.ffchClientService.postUniteEquipements(uniteEquipements)
+      .subscribe(status => this.getEquipmentsFromFfch(), status => console.log("Could not send equipments"));
```

Comment:
Console.log c'est pas le top, this.equipmentErrors.push() serait plus indiqué pour le cas d'erreur je pense.

Reply:
OK